openbis-connector/src/main/java/life/qbic/projectmanagement/persistence/repository/OpenbisConnector.java

```diff
@@ -0,0 +1,86 @@
+package life.qbic.projectmanagement.persistence.repository;
+
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.search.SearchResult;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.vocabulary.VocabularyTerm;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.vocabulary.fetchoptions.VocabularyTermFetchOptions;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.vocabulary.search.VocabularyTermSearchCriteria;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import life.qbic.openbis.openbisclient.OpenBisClient;
+import life.qbic.projectmanagement.domain.project.repository.ExperimentalDesignVocabularyRepository;
+import life.qbic.projectmanagement.domain.project.vocabulary.Analyte;
+import life.qbic.projectmanagement.domain.project.vocabulary.ControlledVocabulary;
+import life.qbic.projectmanagement.domain.project.vocabulary.Species;
+import life.qbic.projectmanagement.domain.project.vocabulary.Specimen;
+import org.springframework.stereotype.Component;
+
+/**
+ * Basic implementation to query project preview information
+ *
+ * @since 1.0.0
+ */
+@Component
+public class OpenbisConnector implements ExperimentalDesignVocabularyRepository {
+
+  final private OpenBisClient openBisClient;
+
+  public OpenbisConnector(OpenBisClient openBisClient) {
+    Objects.requireNonNull(openBisClient);
+    this.openBisClient = openBisClient;
+  }
+
+  private ControlledVocabulary getVocabularyForCode(String vocabularyCode) {
+    VocabularyTermSearchCriteria criteria = new VocabularyTermSearchCriteria();
+    criteria.withVocabulary().withCode().thatEquals(vocabularyCode);
+
+    VocabularyTermFetchOptions options = new VocabularyTermFetchOptions();
+    SearchResult<VocabularyTerm> searchResult =
+        openBisClient.getV3().searchVocabularyTerms(openBisClient.getSessionToken(), criteria, options);
+
+    Map<String,String> termsByName = new HashMap<>();
+    for (VocabularyTerm term : searchResult.getObjects()) {
+      String code = term.getCode();
+      String label = code;
+      if (term.getLabel() != null && !term.getLabel().isEmpty()) {
+        label = term.getLabel();
```

Comment:
Question for clarification: Is the label the value of interest for us? If so then i'd be nitty here and keep the naming consistent between the ControlledVocabulary terminology and this method.